### PR TITLE
Generate ingress based on kubernetes version

### PR DESCRIPTION
**Pull request by leki75**

### CODE REVIEW

File: chart/pyroscope/templates/ingress.yaml [hunk 1]
```diff
@@ -1,7 +1,18 @@
 {{- if .Values.ingress.enabled -}}
 {{- $fullName := include "pyroscope.fullname" . -}}
 {{- $svcPort := .Values.service.port -}}
-apiVersion: {{ .Values.ingress.apiVersion }}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
```
Comment (kolesnikovae) [inline chart/pyroscope/templates/ingress.yaml:4]: What do you think about using `.Capabilities.KubeVersion.Version` (just to avert future breakage: `GitVersion` seems to be [deprecated](https://github.com/helm/helm/blob/5d47255a3d6c886850d6c6cbead1b00b204d07b1/pkg/chartutil/capabilities.go#L83)), or even check API availability via `.Capabilities.APIVersions.Has`?

**Review by kolesnikovae — APPROVED** [review]
Thank you @leki75! This is an awesome improvement, going to merge it.

**Comment by kolesnikovae** [comment]
I'd love to merge the PR but can't due to the conflict. Could you please bump the version and update the docs with helm-docs?

File: chart/pyroscope/templates/ingress.yaml (hunk 1 above)
Comment (leki75) [reply chart/pyroscope/templates/ingress.yaml:4]: I tried to use the same code that `helm create <chart>` provides. It generates the template with `.Capabilities.KubeVersion.GitVersion`, too.

**Comment by kolesnikovae** [comment]
Oh, I'm terribly sorry for the delay. Thank you very much @leki75! Awesome work.

**Pull request merged**